File: packages/react-native-autoplay/ios/hybrid/HybridSearchTemplate.swift
//
//  HybridSearchTemplate.swift
//  Pods
//
//  Created by Samuel Brucksch on 28.10.25.
//

class HybridSearchTemplate: HybridSearchTemplateSpec {
    func createSearchTemplate(config: SearchTemplateConfig) throws {
        let template = SearchTemplate(config: config)
        try RootModule.withScene { scene in
            scene.templateStore.addTemplate(
                template: template,
                templateId: config.id
            )
        }
    }

    func updateSearchResults(templateId: String, results: NitroSection) throws {
        try RootModule.withAutoPlayTemplate(templateId: templateId) {
            (template: SearchTemplate) in
            template.updateSearchResults(results: results)
        }
    }
}

File: packages/react-native-autoplay/android/src/main/java/com/margelo/nitro/swe/iternio/reactnativeautoplay/hybrid/HybridSearchTemplate.kt
package com.margelo.nitro.swe.iternio.reactnativeautoplay

import com.margelo.nitro.swe.iternio.reactnativeautoplay.template.AndroidAutoTemplate
import com.margelo.nitro.swe.iternio.reactnativeautoplay.template.SearchTemplate

class HybridSearchTemplate : HybridSearchTemplateSpec() {

    override fun createSearchTemplate(config: SearchTemplateConfig) {
        val context = AndroidAutoSession.Companion.getRootContext()
            ?: throw IllegalArgumentException("createSearchTemplate failed, carContext not found")

        val template = SearchTemplate(context, config)
        AndroidAutoTemplate.Companion.setTemplate(config.id, template)
    }

    override fun updateSearchResults(templateId: String, results: NitroSection) {
        val template = AndroidAutoTemplate.Companion.getTemplate<SearchTemplate>(templateId)
        template.updateSearchResults(results)
    }
}